.github/workflows/lint.yml

```diff
@@ -28,7 +28,22 @@ jobs:
         with:
           cache-on-failure: true
       - run:
-          cargo clippy --bin "${{ matrix.binary }}" --workspace --features "${{ matrix.network }}" --lib --tests --benches --examples
+          cargo clippy --bin "${{ matrix.binary }}" --workspace --features "${{ matrix.network }} asm-keccak jemalloc jemalloc-prof min-error-logs min-warn-logs min-info-logs min-debug-logs min-trace-logs"
```

Comment:
having one clippy task for the binaries specifically and one for tests etc sounds reasonable

Reply:
this way we can run checks with `--all-features` flag on targets other than the binaries. adding that job makes it easier to maintain. this way we only have to update ci if we add another feature to the binary crates, not the rest of the workspace.